qemu/tests/balloon_hotplug.py

```diff
@@ -111,34 +104,25 @@ def enable_balloon_service():
 
         error_context.context("Unplug balloon device for %d times" % (i+1),
                               logging.info)
-        out = new_dev.unplug(vm.monitor)
-        if out:
-            err += "\nUnplug monitor output: %s" % out
-        # Pause
-        time.sleep(pause)
-        ver_out = new_dev.verify_unplug(out, vm.monitor)
-        if not ver_out:
-            err += ("\nDevice is still in qtree %ss after unplug:\n%s"
-                    % (pause, vm.monitor.info("qtree")))
-        if err:
-            logging.error(vm.monitor.info("qtree"))
-            test.fail("Error occurred while hotpluging "
-                      "virtio-pci. Iteration %s, monitor "
-                      "output:%s" % (i, err))
-        else:
-            if params.get("migrate_after_unplug", "no") == "yes":
-                error_context.context("Migrate after hotunplug balloon device",
-                                      logging.info)
-                # temporary workaround for migration
-                del vm.params["balloon"]
-                del vm.params["balloon_dev_devid"]
-                del vm.params["balloon_dev_add_bus"]
-                vm.migrate(float(params.get("mig_timeout", "3600")))
-
-            if pm_test_after_unplug:
-                run_pm_test(pm_test_after_unplug, "hot-unplug")
-                if not vm.is_alive():
-                    return
+
+        out = vm.devices.simple_unplug(new_dev, vm.monitor)
+        if out[1] is False:
+            test.fail("Failed to unplug balloon in iteration %s, %s"
+                      % (i, out[0]))
+
+        if params.get("migrate_after_unplug", "no") == "yes":
+            error_context.context("Migrate after hotunplug balloon device",
+                                  logging.info)
+            # temporary workaround for migration
+            del vm.params["balloon"]
+            del vm.params["balloon_dev_devid"]
+            del vm.params["balloon_dev_add_bus"]
```

Comment:
Please help check if the above three lines are still needed.

Reply:
Yes, they are needed.  After unplug, guest should not have balloon device. If we remove the lines, dst guest will boot with balloon device even we unplugged it in src.